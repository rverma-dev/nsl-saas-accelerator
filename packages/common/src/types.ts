import TIER from './stacks.json';

export interface DeploymentRecord {
  tenantId?: string; // callhealth
  id: string; // dev1, dev2, prod1, prod2
  type?: string; // can be pool or silo
  tier?: string; // small
  account?: string;
  region?: string;
}

export interface Deployment extends DeploymentRecord {
  provisioned?: boolean;
}
export function isValidTier(tierName: string): boolean {
  return TIER.some(tier => tier.name === tierName);
}

export function getPipelineName(record: DeploymentRecord): string {
  return `${record.tenantId}-${record.id}-${record.type}-pipeline`;
}
