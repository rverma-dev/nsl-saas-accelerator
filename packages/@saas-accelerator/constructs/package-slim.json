{
    "name": "@saas-accelerator/constructs",
    "version": "0.0.1",
    "dependencies": {
        "@jest/globals": "^29.5.0",
        "@aws-quickstart/eks-blueprints": "^1.6.1",
        "aws-cdk-lib": "2.72.0",
        "cdk-nag": "2.25.11",
        "js-yaml": "^4.1.0",
        "sync-request": "^6.1.0"
    },
    "devDependencies": {
        "aws-cdk-lib": "2.72.0",
        "@aws-cdk/assert": "2.68.0",
        "@types/jest": "^29.2.1",
        "@types/node": "18.11.19",
        "constructs": "10.0.12",
        "eslint": "^8.26.0",
        "jest": "^29.2.2",
        "ts-jest": "^29.0.3",
        "typescript": "~5.0.4"
    }
}