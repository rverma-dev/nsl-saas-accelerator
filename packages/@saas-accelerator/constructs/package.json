{
  "name": "@saas-accelerator/constructs",
  "version": "0.0.1",
  "private": true,
  "description": "The cdk constructs library for the accelerator solution.",
  "license": "Apache-2.0",
  "author": {
    "name": "Amazon Web Services",
    "url": "https://aws.amazon.com/solutions"
  },
  "types": "index.d.ts",
  "scripts": {
    "build": "tsc",
    "cleanup": "tsc --build ./ --clean && rm -rf node_modules && rm -rf yarn.lock && rm -rf dist && rm -rf cdk.out",
    "cleanup:tsc": "tsc --build ./ --clean",
    "lint": "eslint --fix  --max-warnings 0 -c ../../../.eslintrc.json 'lib/**/*.{ts,tsx}' 'test/**/*.{ts,tsx}' --ignore-pattern \"*.d.ts\" ",
    "precommit": "eslint --max-warnings 0 -c ../../../.eslintrc.json 'lib/**/*.{ts,tsx}' 'test/**/*.{ts,tsx}' --ignore-pattern \"*.d.ts\" ",
    "test": "jest --coverage --ci",
    "watch": "tsc -w"
  },
  "dependencies": {
    "@jest/globals": "^29.5.0",
    "@aws-quickstart/eks-blueprints": "^1.6.1",
    "aws-cdk-lib": "2.72.0",
    "cdk-nag": "2.25.5",
    "js-yaml": "^4.1.0",
    "sync-request": "^6.1.0"
  },
  "devDependencies": {
    "aws-cdk-lib": "2.72.0",
    "@aws-cdk/assert": "2.68.0",
    "@types/jest": "^29.2.1",
    "@types/node": "18.11.19",
    "constructs": "10.0.12",
    "eslint": "^8.26.0",
    "jest": "^29.2.2",
    "ts-jest": "^29.0.3",
    "typescript": "~5.0.4"
  }
}
