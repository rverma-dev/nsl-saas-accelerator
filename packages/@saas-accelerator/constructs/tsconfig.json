{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {},
  "include": ["lib/*/*.ts", "index.ts"],
  "exclude": ["test/**/*"]
}
