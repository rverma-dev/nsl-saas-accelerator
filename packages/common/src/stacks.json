[
  {
    "name": "small",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "size": "db.small"
    },
    "dynamodb": {
      "size": "14-2"
    }
  },
  {
    "name": "small-rc",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "size": "db.small"
    },
    "dynamodb": {
      "size": "14-2"
    }
  },
  {
    "name": "large",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "size": "db.small"
    },
    "dynamodb": {
      "size": "100-80"
    }
  },
  {
    "name": "serverless",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "maxSize": "2"
    },
    "dynamodb": {
      "maxSize": "14"
    }
  }
]