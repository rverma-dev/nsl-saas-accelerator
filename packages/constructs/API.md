# API Reference
**Classes**
Name|Description
----|-----------
[CreateGitopsSecretResource](#nsa-construct-creategitopssecretresource)|Class to configure CloudWatch Destination on logs receiving account.
[EksCluster](#nsa-construct-ekscluster)|
**Structs**
Name|Description
----|-----------
[CreateGitopsSecretProps](#nsa-construct-creategitopssecretprops)|
[EKSClusterProps](#nsa-construct-eksclusterprops)|


## class CreateGitopsSecretResource  <a id="nsa-construct-creategitopssecretresource"></a>

Class to configure CloudWatch Destination on logs receiving account.

**Implements**: [IConstruct](#constructs-iconstruct), [IDependable](#constructs-idependable)
**Extends**: [Construct](#constructs-construct)

### Initializer




```
new CreateGitopsSecretResource(scope: Construct, id: string, props: CreateGitopsSecretProps)
```

* **scope** (<code>[Construct](#constructs-construct)</code>)  *No description*
* **id** (<code>string</code>)  *No description*
* **props** (<code>[CreateGitopsSecretProps](#nsa-construct-creategitopssecretprops)</code>)  *No description*
  * **secretName** (<code>string</code>)  Secret name. 
  * **username** (<code>string</code>)  Gitops IAM username. 



### Properties


Name | Type | Description 
-----|------|-------------
**id** | <code>string</code> | <span></span>
**node** | <code>[Node](#constructs-node)</code> | The tree node.

### Methods


Name | Description
-----|-----
[**toString()**](#nsa-construct-creategitopssecretresource-tostring) | Returns a string representation of this construct.


---
#### toString() <a id="nsa-construct-creategitopssecretresource-tostring"></a>

Returns a string representation of this construct.
```
public toString(): string
```


*Returns*
* <code>string</code>



## class EksCluster  <a id="nsa-construct-ekscluster"></a>



**Implements**: [IConstruct](#constructs-iconstruct), [IDependable](#constructs-idependable)
**Extends**: [Construct](#constructs-construct)

### Initializer




```
new EksCluster(scope: Construct, props: EKSClusterProps)
```

* **scope** (<code>[Construct](#constructs-construct)</code>)  *No description*
* **props** (<code>[EKSClusterProps](#nsa-construct-eksclusterprops)</code>)  *No description*
  * **platformTeamRole** (<code>string</code>)  *No description* 
  * **gitopsRepoBranch** (<code>string</code>)  *No description* *Optional*
  * **gitopsRepoSecret** (<code>string</code>)  *No description* *Optional*
  * **gitopsRepoUrl** (<code>string</code>)  *No description* *Optional*
  * **vpcID** (<code>string</code>)  *No description* *Optional*



### Properties


Name | Type | Description 
-----|------|-------------
**node** | <code>[Node](#constructs-node)</code> | The tree node.

### Methods


Name | Description
-----|-----
[**toString()**](#nsa-construct-ekscluster-tostring) | Returns a string representation of this construct.


---
#### toString() <a id="nsa-construct-ekscluster-tostring"></a>

Returns a string representation of this construct.
```
public toString(): string
```


*Returns*
* <code>string</code>



## struct CreateGitopsSecretProps  <a id="nsa-construct-creategitopssecretprops"></a>






Name | Type | Description 
-----|------|-------------
**secretName** | <code>string</code> | Secret name.
**username** | <code>string</code> | Gitops IAM username.



## struct EKSClusterProps  <a id="nsa-construct-eksclusterprops"></a>






Name | Type | Description 
-----|------|-------------
**platformTeamRole** | <code>string</code> | <span></span>
**gitopsRepoBranch**? | <code>string</code> | <br/><br/>*Optional*
**gitopsRepoSecret**? | <code>string</code> | <br/><br/>*Optional*
**gitopsRepoUrl**? | <code>string</code> | <br/><br/>*Optional*
**vpcID**? | <code>string</code> | <br/><br/>*Optional*



