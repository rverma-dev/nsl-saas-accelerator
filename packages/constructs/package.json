{
  "name": "@nsa/construct",
  "repository": {
    "type": "git",
    "url": "https://github.com/rverma-nsl/nsl-saas-accelerator.git"
  },
  "scripts": {
    "build": "npx projen build",
    "bundle": "npx projen bundle",
    "bundle:aws-codecommit/secret-gen.lambda": "npx projen bundle:aws-codecommit/secret-gen.lambda",
    "bundle:aws-codecommit/secret-gen.lambda:watch": "npx projen bundle:aws-codecommit/secret-gen.lambda:watch",
    "compat": "npx projen compat",
    "compile": "npx projen compile",
    "default": "npx projen default",
    "docgen": "npx projen docgen",
    "eslint": "npx projen eslint",
    "package": "npx projen package",
    "package-all": "npx projen package-all",
    "package:js": "npx projen package:js",
    "post-compile": "npx projen post-compile",
    "post-upgrade": "npx projen post-upgrade",
    "pre-compile": "npx projen pre-compile",
    "test": "npx projen test",
    "test:watch": "npx projen test:watch",
    "upgrade": "npx projen upgrade",
    "watch": "npx projen watch",
    "projen": "npx projen"
  },
  "author": {
    "name": "Rohit",
    "email": "rohit.verma@nslhub.com",
    "organization": false
  },
  "devDependencies": {
    "@types/aws-lambda": "^8.10.114",
    "@types/jest": "^29.1.0",
    "@types/node": "^18",
    "@typescript-eslint/eslint-plugin": "^5",
    "@typescript-eslint/parser": "^5",
    "aws-cdk-lib": "2.76.0",
    "aws-lambda": "^1.0.7",
    "cdk-nag": "2.26.1",
    "constructs": "10.2.4",
    "esbuild": "^0.0.0",
    "eslint": "^8",
    "eslint-import-resolver-node": "^0.3.7",
    "eslint-import-resolver-typescript": "^3.5.5",
    "eslint-plugin-import": "^2.27.5",
    "jest": "^29.1.0",
    "jest-junit": "^15",
    "jsii": "5.0.6",
    "jsii-diff": "^0.8.2",
    "jsii-docgen": "^0.0.0",
    "jsii-pacmak": "1.80.0",
    "npm-check-updates": "^16",
    "projen": "^0.71.19",
    "ts-jest": "^29.1.0",
    "typescript": "^5.0.4"
  },
  "peerDependencies": {
    "aws-cdk-lib": "^2.76.0",
    "cdk-nag": "^2.26.1",
    "constructs": "^10.2.4"
  },
  "dependencies": {
    "@aws-sdk/client-iam": "^3.316.0",
    "@aws-sdk/client-secrets-manager": "^3.316.0",
    "@nslhb/eks-blueprints": "1.6.21",
    "@pnpm/reviewing.dependencies-hierarchy": "*",
    "@types/fs-extra": "11.0.1",
    "@types/js-yaml": "4.0.5",
    "@types/semver": "7.3.13",
    "js-yaml": "4.1.0",
    "sync-request": "^6.1.0"
  },
  "bundledDependencies": [
    "@aws-sdk/client-iam",
    "@aws-sdk/client-secrets-manager",
    "@nslhb/eks-blueprints",
    "@pnpm/reviewing.dependencies-hierarchy",
    "@types/fs-extra",
    "@types/js-yaml",
    "@types/semver",
    "js-yaml",
    "sync-request"
  ],
  "pnpm": {},
  "keywords": [
    "cdk"
  ],
  "engines": {
    "node": ">= 18.0.0"
  },
  "main": "lib/index.js",
  "license": "Apache-2.0",
  "version": "0.0.0",
  "jest": {
    "detectOpenHandles": true,
    "testMatch": [
      "<rootDir>/src/**/__tests__/**/*.ts?(x)",
      "<rootDir>/(test|src)/**/*(*.)@(spec|test).ts?(x)"
    ],
    "clearMocks": true,
    "collectCoverage": true,
    "coverageReporters": [
      "json",
      "lcov",
      "clover",
      "cobertura",
      "text"
    ],
    "coverageDirectory": "coverage",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ],
    "testPathIgnorePatterns": [
      "/node_modules/"
    ],
    "watchPathIgnorePatterns": [
      "/node_modules/"
    ],
    "reporters": [
      "default",
      [
        "jest-junit",
        {
          "outputDirectory": "test-reports"
        }
      ]
    ],
    "preset": "ts-jest",
    "globals": {
      "ts-jest": {
        "tsconfig": "tsconfig.dev.json"
      }
    }
  },
  "types": "lib/index.d.ts",
  "stability": "stable",
  "jsii": {
    "outdir": "dist",
    "targets": {},
    "tsc": {
      "outDir": "lib",
      "rootDir": "src"
    }
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"npx projen\"."
}
