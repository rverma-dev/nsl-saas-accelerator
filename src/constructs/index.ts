export * from './aws-codecommit';
export * from './aws-eks';
export * from './cdk-pipeline';
