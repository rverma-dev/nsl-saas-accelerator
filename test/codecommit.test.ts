import { CreateGitopsSecretResource } from '../src/constructs';
import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();
describe('GitopsSecret', () => {
  test('Snapshot test', () => {
    new CreateGitopsSecretResource(stack, 'codeCommitSecret', {
      username: 'XXXX',
      secretName: 'demo',
    });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
