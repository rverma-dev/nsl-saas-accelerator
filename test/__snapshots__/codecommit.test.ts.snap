// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitopsSecret Construct(GitopsStack):  Snapshot Test 1`] = `[Function]`;
